File: src/ContableUxser.Domain/Entities/Venta.cs
using ContableUxser.Domain.Common;
using ContableUxser.Domain.Enums;

namespace ContableUxser.Domain.Entities;

public class Venta : BaseEntity
{
    public Guid EmpresaId { get; set; }
    public Guid SesionCajaId { get; set; }
    public Guid UsuarioId { get; set; }
    public DateTime FechaVenta { get; set; } = DateTime.UtcNow;
    public decimal Total { get; set; }
    public MetodoPago MetodoPago { get; set; }
    public string? ReferenciaTransferencia { get; set; }
    public bool SincronizadoNube { get; set; }

    public Empresa Empresa { get; set; } = null!;
    public SesionCaja SesionCaja { get; set; } = null!;
    public Usuario Usuario { get; set; } = null!;
    public ICollection<VentaDetalle> VentaDetalles { get; set; } = new List<VentaDetalle>();
}

File: contable_uxser_app/lib/domain/entities/venta.dart
import '../enums/metodo_pago.dart';

class Venta {
  final String id;
  final String empresaId;
  final String sesionCajaId;
  final String usuarioId;
  final DateTime fechaVenta;
  final double total;
  final MetodoPago metodoPago;
  final String? referenciaTransferencia;
  final bool sincronizadoNube;
  final List<VentaDetalle> detalles;

  Venta({
    required this.id,
    required this.empresaId,
    required this.sesionCajaId,
    required this.usuarioId,
    required this.fechaVenta,
    required this.total,
    required this.metodoPago,
    this.referenciaTransferencia,
    this.sincronizadoNube = false,
    this.detalles = const [],
  });
}

class VentaDetalle {
  final String id;
  final String ventaId;
  final String productoId;
  final String? productoNombre;
  final double cantidad;
  final double precioUnitario;
  final double subtotal;

  VentaDetalle({
    required this.id,
    required this.ventaId,
    required this.productoId,
    this.productoNombre,
    required this.cantidad,
    required this.precioUnitario,
    required this.subtotal,
  });
}
